### ISSUE TITLE
incompatible with vite

### ISSUE THREAD

**Issue by weijarz** [danbovey/react-infinite-scroller#253]
**Describe the bug**

Incompatible with vite:
https://github.com/vitejs/vite/issues/527#issuecomment-654908920

**Comment by pcdv** [comment]
Probably due to the presence of `src` in .npmignore

**Comment by jmgirven** [comment]
I am currently getting round the problem by editing `node_modules/react-infinite-scroller/package.json` and changing:

`"jsnext:main": "src/InfiniteScroll.js",`

to:

`"jsnext:main": "dist/InfiniteScroll.js",`

Either (like pcdv said) the library should ship with the src directory, or make the above modification to package.json.

**Comment by pcdv** [comment]
I'm not a JS expert but I think the spirit of `jsnext:main` is to point to the original source, without babel transpilation (so modern browsers can use it directly). So the correct solution would be to point to the original source in `jsnext:main`

Pointing to the transpiled file would work too but in that case, we might as well remove the `jsnext:main` tag altogether because it would bring no additional value.

**Comment by ghernandezr** [comment]
Hello. I have the same issue. Any solution so far?
Thanks!!!!

**Comment by qWici** [comment]
Any updates?

**Comment by Eva-HYL** [comment]
I have the same issue.   Any updates?

**Comment by qWici** [comment]
> I have the same issue. Any updates?

If you want a simple version, I wrote mine. The code is [here](https://gist.github.com/qWici/70bcf97d697148c323042a0b3395fbed)

**Comment by RickieWoo** [comment]
so nobody is fixing this issue?
it's a little bit stupid that I have to modify the package.json every time 😅

**Comment by wmzy** [comment]
```ts
export default defineConfig({
  resolve: {
    alias: {
      'react-infinite-scroller': 'react-infinite-scroller/index.js'
    },
  }
});
```

**Comment by danbovey** [comment]
Fixed by #279

**Comment by mqliutie** [comment]
> ```ts
> export default defineConfig({
>   resolve: {
>     alias: {
>       'react-infinite-scroller': 'react-infinite-scroller/index.js'
>     },
>   }
> });
> ```

`No matching export in "node_modules/react-virtualized/dist/es/WindowScroller/WindowScroller.js" for import "bpfrpt_proptype_WindowScroller"`

I added this alias but new error is shown
